test/DEVELOPERS.md

```diff
@@ -0,0 +1,879 @@
+# Developer documentation
+
+## OC, BC and Dual NIC BC configuration Models
+The following ptp clocks type are used in this documentation: Ordinary Clock (OC), Boundary Clock (BC) and Dual Nic Boundary Clock (Dual NIC BC). PTP clocks are useful for hosts that needs to synchronize their hardware clock very accurately to around few nanoseconds.
+
+### Ordinary Clock
+The simplest ptp clock configuration is the Ordinary clock. In this case, the clock just needs to received synchronization info from a source of synchronization or master. It is very similar in functionality to the less accurate but ubiquitous Network Time Protocol (NTP), just setting the Host internal clock to track a very accurate master clock. The root source of synchronization is the Grand master. It is determined by the Best Master Clock Algorithm. 
+
+For the purpose of this test suite, we consider that clock configuration is static, so the Grandmaster will not move because of the very constrained scenario. Cases that consider Best Master Clock algorithm could be created in the future.
+
+`ptp4l` is the executable responsible for the ptp clock synchronization. ptp4l role is to synchronize the NIC's PTP hardware clock (PHC) with the remote master clock. 
+
+`phc2sys` is another executable responsible for synchronizing the PHC clock to the host's OS clock 
+
+![overview_oc](doc/ocoverview.svg)
+
+### Boundary Clock
+Boundary clocks (BC) also synchronize to a master clock , just like an ordinary clock. In addition, BC clocks are also a master to downstream ordinary or boundary clocks. This allow propagating synchronization information to many hosts. Network switches are great location to implement boundary clock since they connect to many hosts. However, a regular server equipped with a 2-ports network card supporting PTP will also be able to implement PTP BC clocks.
+
+![overview_bc](doc/bcoverview.svg)
+
+### Dual NIC Boundary Clock
+Dual NIC BC are similar to boundary clocks except that their provide more master ports to synchronize more ordinary/boundary clocks. Network cards are usually limited to a maximum of 4 ports sharing a PTP Hardware clock. If more than 4 downstream clocks need to received synchronization 2 NIC can be configured together using the Dual NIC BC pattern. 
+
+A primary Boundary clock receives synchronization from a master clock and a secondary boundary clock also receives updates from a Master clock which synchronizes its PHC. However, the secondary BC does not run phc2sys to synchronize its PHC with the OS clock. If the Primary connection to the Grandmaster is lost, the secondary BC acts as the primary.
```

Comment:
If the Primary connection to the Grandmaster is lost, the secondary BC acts as the primary. We don't run this as HA. so both will be synching differently downstream. if the primary is lost, The secondary won't sync primary PHC

Reply:
fixed